src/pynwb/tests/unit/test_ernwbfile2.py

```diff
@@ -92,3 +100,72 @@ def test_custom_type_map(self):
         type_map = TypeMap()
         er = NWBExternalResources('ER', type_map=type_map)
         self.assertIs(er.type_map, type_map)
+
+
+class TestTaskSeriesRoundtrip(TestCase):
+    """Simple roundtrip test for ER."""
+
+    def setUp(self):
+        self.nwbfile = set_up_nwbfile()
+        self.path = "test.nwb"
+
+    def tearDown(self):
+        remove_test_file(self.path)
+
+    def test_roundtrip(self):
+        nwbfile = ERNWBFile(
+            session_description='session_description',
+            identifier='identifier',
+            session_start_time=datetime.datetime.now(datetime.timezone.utc)
+        )
+
+        container = TimeSeries(
+            name='test_ts',
+            data=[1, 2, 3],
+            unit='meters',
+            timestamps=[0.1, 0.2, 0.3],
+        )
+        nwbfile.add_acquisition(container)
+
+        table = DynamicTable(name='test_table', description='test table description')
+        table.add_column(name='test_col', description='test column description')
+        table.add_row(test_col='Mouse')
+
+        nwbfile.add_acquisition(table)
+
+        nwbfile.external_resources.add_ref(
+            container=container,
+            field='unit',
+            key='meters',
+            resource_name='SI_Ontology',
+            resource_uri='',
+            entity_id='5',
+            entity_uri='',
+        )
+
+        nwbfile.external_resources.add_ref(
+            container=table,
+            attribute='test_col',
+            key='Mouse',
+            resource_name='NCBI_Taxonomy',
+            resource_uri='https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi',
+            entity_id='10090',
+            entity_uri='https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=info&id=10090',
+        )
+
+        path = 'test.nwb'
+        with NWBHDF5IO(path, mode='w') as io:
+            io.write(nwbfile)
+
+        with NWBHDF5IO(path, mode='r', load_namespaces=True) as io:
+            read_nwbfile = io.read()
+            self.assertEqual(len(read_nwbfile.external_resources.objects), 2)
+            self.assertEqual(len(read_nwbfile.external_resources.object_keys), 2)
+            self.assertEqual(len(read_nwbfile.external_resources.keys), 2)
+            self.assertEqual(len(read_nwbfile.external_resources.resources), 2)
+            self.assertEqual(len(read_nwbfile.external_resources.entities), 2)
+
+            self.assertEqual(list(read_nwbfile.external_resources.objects[0]),
+                             list(nwbfile.external_resources.objects[0]))
+            self.assertEqual(list(read_nwbfile.external_resources.objects[1]),
+                             list(nwbfile.external_resources.objects[1]))
```

Comment:
Can you also test and add `self.assertContainerEqual(read_nwbfile.external_resources, nwbfile.external_resources)`? It will compare all the children recursively and comprehensively. You should be able to replace the last two `self.assertEqual` lines here with that.